Tools/WinMLRunner/BindingUtilities.h

```diff
@@ -144,6 +144,35 @@ namespace BindingUtilities
         return elementStrings;
     }
 
+    template <typename T>
+    void PreProcessImageToBinding(const SoftwareBitmap& softwareBitmap, ModelBinding<T>& binding, float scale, const std::array<float, 3>& meanStdDev)
+    {
+        const auto imgHeight = softwareBitmap.PixelHeight();
+        const auto imgWidth = softwareBitmap.PixelWidth();
+        const auto resultSize = imgHeight * imgWidth * 3;
+        if (binding.GetDataBufferSize() != resultSize)
+        {
+            throw hresult_invalid_argument(L"Image Input size/shape is different from what model expects");
+        }
+
+        T* bindingData = binding.GetData();
+        if (scale != 1.0f &&
+            (meanStdDev[0] != 0 || meanStdDev[1] != 0 || meanStdDev[2] != 0))
+        {
+            Buffer sbBuffer(imgHeight * imgWidth * 4);
+            softwareBitmap.CopyToBuffer(sbBuffer);
+            uint8_t *sbBufferData = sbBuffer.data();
+
+            //Roll the array correctly for the tensor
+            for (int i = 0, count = 0; i < imgHeight * imgWidth; ++i, count += 4)
+            {
+                bindingData[i] = (sbBufferData[count] - meanStdDev[0]) / scale;
```

Comment:
Can you do an explicit cast here?  
(T)(sbBufferData[count] - meanStdDev[0]) / scale;

Reply:
Yes, I will do this with my next commit.